### ISSUE TITLE
concatMapSingle ignores some onNext events

### ISSUE THREAD

**Issue by acadet** [ReactiveX/RxJava#5971]
**Version used: 2.1.12**

I have a collection of interfaces. They all define a method fetching a collection of items: `Single<List<Integer>> fetchItems()`. 
Some of them perform local computations, others a network call. When concatening using `concatMapSingle`, some of the singles are not subscribed at all.

Here is a test I do think should pass. A network call is faked here using a timer. If the timer is replaced with `Single.just` instead, the test is passing.

```java
@Test
public void test() {
  TestScheduler testScheduler = new TestScheduler();

  final Single<List<Integer>> first = Single.timer(2, TimeUnit.SECONDS, testScheduler).map(u -> Arrays.asList(1, 2, 3));
  final Single<List<Integer>> second = Single.just(Collections.emptyList());
  final Single<List<Integer>> third = Single.just(Collections.singletonList(4));
  final Single<List<Integer>> fourth = Single.just(Collections.singletonList(5));

  Single<List<Integer>> subject = Observable
    .fromIterable(Arrays.asList(first, second, third, fourth))
    .concatMapSingle(single -> single)
    .reduce(new ArrayList<>(), (seed, items) -> {
      seed.addAll(items);
      return seed;
    });

    TestObserver<List<Integer>> testObserver = subject.test();
    testScheduler.advanceTimeBy(2, TimeUnit.SECONDS);

    testObserver.assertValue(list -> list.equals(Arrays.asList(1, 2, 3, 4, 5))); 
    // 5 is currently missing ; fourth was never subscribed in the first place
}
```

*It is my first issue on this repo, feel free to edit my PR if I am not respecting the contribution guidelines.*

**Comment by akarnokd** [comment]
Hi and thanks for reporting. This is a bug in the `concatMapSingle` operator. I'll post a fix shortly.

**Comment by akarnokd** [comment]
Fix in #5972.

**Comment by acadet** [comment]
Thanks, appreciate it!
